src/extensions/mvs/tree/generic/tree-schema.ts

```diff
@@ -10,14 +10,19 @@ import { AllRequired, DefaultsFor, ParamsSchema, ValuesFor, paramsValidationIssu
 import { treeToString } from './tree-utils';
 
 
+/** Type of additional_properties of a tree node (key-value storage with arbitrary JSONable values) */
+export type AdditionalProperties = Partial<Record<string, any>>
+
 /** Tree node without children */
 export type Node<TKind extends string = string, TParams extends {} = {}> =
     {} extends TParams ? {
         kind: TKind,
-        params?: TParams,
+        params: TParams | undefined,
+        additional_properties: AdditionalProperties | undefined,
     } : {
         kind: TKind,
         params: TParams,
+        additional_properties: AdditionalProperties | undefined,
```

Comment:
```suggestion
        additional_properties?: AdditionalProperties,
```

I think we could go the optional route here so you don't have explicitly specify undefined everywhere.